### PR TITLE
use unique_ptr for packets in utp_stream

### PR DESCRIPTION
**Pull request by arvidn**
@ajax16384 does this make sense? just pulling out to use ``packet_ptr`` in the utp implementation too

### CODE REVIEW

**Comment by ajax16384** [comment]
ok

File: src/packet_buffer.cpp [hunk 1]
```diff
@@ -53,15 +53,15 @@ namespace libtorrent {
...
+	packet_ptr packet_buffer::insert(index_type idx, packet_ptr value)
 	{
 		INVARIANT_CHECK;
 
 		TORRENT_ASSERT_VAL(idx <= 0xffff, idx);
 		// you're not allowed to insert NULLs!
 		TORRENT_ASSERT(value);
 
-		if (value == nullptr) return remove(idx);
+		if (!value) return remove(idx);
```
Comment (aldenml) [inline src/packet_buffer.cpp]: @arvidn I have a question here: a couple of lines above, there is an assert for null `TORRENT_ASSERT(value);`, is it OK, valid?

Comment (arvidn) [inline src/packet_buffer.cpp]: I don't think you can compare a ``unique_ptr`` against a raw pointer or ``nullptr`` directly. But it has a bool conversion operator

Comment (arvidn) [inline src/packet_buffer.cpp]: I see what you mean. yes, I believe the call to ``remove()`` is defensive here

**Pull request merged**